### PR TITLE
add authorized creators for group reward distributors

### PR DESCRIPTION
**Pull request by soroshansari**

### CODE REVIEW

File: programs/cardinal-group-reward-distributor/src/instructions/init_group_reward_distributor.rs [hunk 1]
```diff
@@ -63,6 +64,7 @@ pub fn handler<'key, 'accounts, 'remaining, 'info>(ctx: Context<'key, 'accounts,
     group_reward_distributor.metadata_kind = GroupRewardDistributorMetadataKind::from(ix.metadata_kind);
     group_reward_distributor.pool_kind = GroupRewardDistributorPoolKind::from(ix.pool_kind);
     group_reward_distributor.authorized_pools = ix.authorized_pools;
+    group_reward_distributor.authorized_creators = ix.authorized_creators;
```
Comment (gchatz22) [inline programs/cardinal-group-reward-distributor/src/instructions/init_group_reward_distributor.rs:67]: we should add resizing of the account in the end of this instruction to account for the vector, same way you did for the respective update ix

Comment (soroshansari) [reply programs/cardinal-group-reward-distributor/src/instructions/init_group_reward_distributor.rs:67]: I've updated the initial size for init, which should work fine.

Comment (soroshansari) [reply programs/cardinal-group-reward-distributor/src/instructions/init_group_reward_distributor.rs:67]: Also, previously in another contract, I couldn't resize the account during the init. Have you tried that yet?

Comment (gchatz22) [reply programs/cardinal-group-reward-distributor/src/instructions/init_group_reward_distributor.rs:67]: yah heres an example of our v2 version of staking https://gist.github.com/gchatz22/3a1f2ac7c34364052e4d4435afeca815

Comment (soroshansari) [reply programs/cardinal-group-reward-distributor/src/instructions/init_group_reward_distributor.rs:67]: Cool, I'll look into it.

**Pull request merged**